src-tauri/src/commands.rs

```diff
@@ -18,14 +18,38 @@ use crate::{
 
 pub mod venv;
 
-/// Measure the current system CPU usage. (unused, for future reference)
+/// Measure the kernel's CPU and memory usage as a percentage of total system
+/// resources.
 #[tauri::command]
-pub async fn cpu_usage() -> f32 {
-    let mut system = System::new();
-    system.refresh_cpu();
+pub async fn kernel_usage_info(
+    kernel_id: &str,
+    _state: tauri::State<'_, State>,
+) -> Result<(f32, f32), Error> {
+    // find the pid from _state.kernels
+    let kernel = _state.kernels.get(kernel_id).ok_or(Error::KernelNotFound)?;
```

Comment:
nit: usually we wouldn't use underline in front of variable name unless it's an unused variable